src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1713,14 +1713,31 @@ def validate_vmss_disk(cmd, namespace):
         raise CLIError('usage error: --disk EXIST_DISK --instance-id ID')
 
 
-def process_disk_or_snapshot_create_namespace(cmd, namespace):
+def process_disk_create_namespace(cmd, namespace):
+    from azure.core.exceptions import HttpResponseError
+    validate_tags(namespace)
+    validate_edge_zone(cmd, namespace)
+    if namespace.source:
+        usage_error = 'usage error: --source {SNAPSHOT | DISK | RESTOREPOINT} | ' \
+                      '--source VHD_BLOB_URI [--source-storage-account-id ID]'
+        try:
+            namespace.source_blob_uri, namespace.source_disk, namespace.source_snapshot, \
+                namespace.source_restore_point, _ = _figure_out_storage_source(
+                    cmd.cli_ctx, namespace.resource_group_name, namespace.source)
+            if not namespace.source_blob_uri and namespace.source_storage_account_id:
+                raise CLIError(usage_error)
```

Comment:
Could you please use a specific error type instead of CLIError?

Reply:
In fact, I split `process_disk_or_snapshot_create_namespace` into `process_disk_create_namespace` and `process_snapshot_create_namespace` due to `--source` should support DiskRestorePoint Id only when creating a disk. `CLIError` was formerly used in `process_disk_or_snapshot_create_namespace`. 
I will replace `CLIError` with `ArgumentUsageError`.